Can't retrieve skull texture in 1.20.4

### Just making sure

- [X] I made sure, this bug is not already known or reported.
- [X] I believe, that this bug is fixable.

### Reproduction

In 1.20.4 the `Property` class in the AuthLib was changed to a record. This results in the value accessor method not being named `getValue()` anymore - but rather the record access style `value()`.

```
java.util.concurrent.CompletionException: java.lang.NoSuchMethodError: 'java.lang.String com.mojang.authlib.properties.Property.getValue()'
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:722) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at com.plotsquared.bukkit.util.task.BukkitPlotSquaredTask.runTask(BukkitPlotSquaredTask.java:39) ~[plotsquared-bukkit-7.3.1-SNAPSHOT.jar:?]
	at com.plotsquared.core.util.task.PlotSquaredTask.run(PlotSquaredTask.java:44) ~[plotsquared-bukkit-7.3.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R3.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.20.4.jar:git-Paper-365]
	at org.bukkit.craftbukkit.v1_20_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:480) ~[paper-1.20.4.jar:git-Paper-365]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1639) ~[paper-1.20.4.jar:git-Paper-365]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:446) ~[paper-1.20.4.jar:git-Paper-365]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1518) ~[paper-1.20.4.jar:git-Paper-365]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1214) ~[paper-1.20.4.jar:git-Paper-365]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:321) ~[paper-1.20.4.jar:git-Paper-365]
	at java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.NoSuchMethodError: 'java.lang.String com.mojang.authlib.properties.Property.getValue()'
	at eu.decentsoftware.holograms.api.utils.items.SkullUtils.getSkullTexture(SkullUtils.java:69) ~[DecentHolograms-2.8.6.jar:?]
	at eu.decentsoftware.holograms.api.utils.items.ItemBuilder.getSkullTexture(ItemBuilder.java:309) ~[DecentHolograms-2.8.6.jar:?]
	at eu.decentsoftware.holograms.api.utils.items.HologramItem.fromItemStack(HologramItem.java:138) ~[DecentHolograms-2.8.6.jar:?]
	at eu.decentsoftware.holograms.api.DHAPI.insertHologramLine(DHAPI.java:496) ~[DecentHolograms-2.8.6.jar:?]
	at eu.decentsoftware.holograms.api.DHAPI.insertHologramLine(DHAPI.java:453) ~[DecentHolograms-2.8.6.jar:?]
	at com.plotsquared.holoplots.provider.impl.DecentHologramsProvider.updateHologram(DecentHologramsProvider.java:48) ~[HoloPlots-7.0.1-SNAPSHOT.jar:?]
	at com.plotsquared.holoplots.provider.HologramProvider.createOrUpdateHologram(HologramProvider.java:70) ~[HoloPlots-7.0.1-SNAPSHOT.jar:?]
	at com.plotsquared.holoplots.DefaultHoloPlots.lambda$updatePlot$3(DefaultHoloPlots.java:188) ~[HoloPlots-7.0.1-SNAPSHOT.jar:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718) ~[?:?]
	... 11 more
```

### Solution

https://github.com/DecentSoftware-eu/DecentHolograms/blob/main/src/main/java/eu/decentsoftware/holograms/api/utils/items/SkullUtils.java#L69

When accessing the value, the server's version should be respected and if required the new method called. Possibly one implementation using reflection and one to directly access the method. ~~In terms of LTS, the "old" `getValue()` should be accessed with reflection, whereas the authlib dependency should be updated and `value()` called directly.~~ AuthLib can't be upgraded as the dependency requires Java 17, whereas DH is stuck on 8. So I guess newer version have to live with reflections (which is quite a pity)

### Server Version

git-Paper-365 (MC: 1.20.4)

### Client Version

1.20.4

### Plugin Version

v2.8.6

### Log

-

--- a/src/main/java/eu/decentsoftware/holograms/api/utils/items/SkullUtils.java
+++ b/src/main/java/eu/decentsoftware/holograms/api/utils/items/SkullUtils.java
@@ -6,6 +6,7 @@ import com.mojang.authlib.properties.PropertyMap;
 import eu.decentsoftware.holograms.api.utils.reflect.Version;
 import lombok.NonNull;
 import lombok.experimental.UtilityClass;
+import org.bukkit.Bukkit;
 import org.bukkit.SkullType;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.meta.ItemMeta;
@@ -18,12 +19,15 @@ import org.json.simple.parser.JSONParser;
 import java.io.BufferedReader;
 import java.io.InputStreamReader;
 import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.net.URLConnection;
 import java.util.Collection;
 import java.util.UUID;
+import java.util.function.Function;
+import java.util.logging.Level;
 
 /**
  * Utility class for modifying the textures or owners or skull ItemStacks.
@@ -38,6 +42,9 @@ public final class SkullUtils {
 	private static Field PROFILE_FIELD;
 	private static Method SET_PROFILE_METHOD;
 	private static boolean INITIALIZED = false;
+
+	private static Method PROPERTY_VALUE_METHOD;
+	private static Function<Property, String> VALUE_RESOLVER;
 
 	/**
 	 * Get the Base64 texture of the given skull ItemStack.
@@ -63,13 +70,33 @@ public final class SkullUtils {
 				return null;
 			}
 
+			if (VALUE_RESOLVER == null) {
+				try {
+					// Pre 1.20(.4?) uses getValue
+					Property.class.getMethod("getValue");
+					VALUE_RESOLVER = Property::getValue;
+				} catch (NoSuchMethodException ignored) {
+					// Since 1.20(.4?) the Property class is a record and utilizes record-style getter methods
+					//noinspection JavaReflectionMemberAccess - method does exist in newer versions
+					PROPERTY_VALUE_METHOD = Property.class.getMethod("value");
+					VALUE_RESOLVER = property -> {
+						try {
+							return (String) PROPERTY_VALUE_METHOD.invoke(property);
+						} catch (IllegalAccessException | InvocationTargetException e) {
+							Bukkit.getLogger().log(Level.SEVERE, "[DecentHolograms] Failed to invoke Property#value", e);
+						}
+						return null;
+					};
+				}
+			}
+
 			PropertyMap properties = profile.getProperties();
 			Collection<Property> property = properties.get("textures");
 			if (property != null && !property.isEmpty()) {
-				return property.stream().findFirst().get().getValue();
+				return VALUE_RESOLVER.apply(property.iterator().next());
 			}
 		} catch (Exception e) {
-			e.printStackTrace();
+			Bukkit.getLogger().log(Level.SEVERE, "[DecentHolograms] Unhandled exception while retrieving skull texture", e);
 		}
 		return null;
 	}

